### PR TITLE
Update example for conditions to use SSA go tags

### PR DESCRIPTION
**Pull request by jpbetz**
**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/autoscaler/pull/5848#discussion_r1297485877

### CODE REVIEW

**Comment by jpbetz** [comment]
cc @liggitt

File: contributors/devel/sig-architecture/api-conventions.md [hunk 1]
```diff
@@ -428,6 +428,15 @@ It should be included as a top level element in status, similar to
 Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
```
Comment (liggitt) [inline contributors/devel/sig-architecture/api-conventions.md:428]: do we need the listType / listMapKey comments in built-in types as well, or is the patchStrategy/patchMergeKey inherited by SSA?

Comment (jpbetz) [reply contributors/devel/sig-architecture/api-conventions.md:428]: "If listType is missing, the API server interprets a patchMergeStrategy=merge marker as a listType=map and the corresponding patchMergeKey marker as a listMapKey." ([xref](https://kubernetes.io/docs/reference/using-api/server-side-apply/))

That said, it looks like Conditions for built-in types are often defined as:

```
         // +listType=map                                                                                                           
         // +listMapKey=type
```

I can update this doc for that as well.

Comment (liggitt) [reply contributors/devel/sig-architecture/api-conventions.md:428]: yeah, if we do that, that works for built-in and kubebuilder types, so we can drop the kubebuilder language

Comment (jpbetz) [reply contributors/devel/sig-architecture/api-conventions.md:428]: @apelisse is there any reason to recommend `+patchStrategy` and `+patchMergeKey` on new built-in types? I see a mix of "only patchStrategy/patchMergeKey", "both patchStrategy/patchMergeKey and SSA listType/listMapKey" and "only  SSA listType/listMapKey" in the built-in types.

Comment (jpbetz) [reply contributors/devel/sig-architecture/api-conventions.md]: simplified to just use SSA in the built-in example form

Comment (liggitt) [reply contributors/devel/sig-architecture/api-conventions.md]: Well… don't we want CSA to still work?

Comment (jpbetz) [reply contributors/devel/sig-architecture/api-conventions.md]: Found another patch strategy ref in the doc. I've updated that too.

Comment (jpbetz) [reply contributors/devel/sig-architecture/api-conventions.md]: > Well… don't we want CSA to still work?

I'm not sure.  We have a couple cases like: https://github.com/kubernetes/kubernetes/blob/f8b5f1a77bbab217159b1d024000770f8c95ed97/staging/src/k8s.io/api/flowcontrol/v1beta2/types.go#L360

Comment (jpbetz) [reply contributors/devel/sig-architecture/api-conventions.md]: Should we audit all of this?   I'm seeing a wide range of annotation patterns. There are some surprising things in here like [JobCondition](https://github.com/kubernetes/kubernetes/blob/f8b5f1a77bbab217159b1d024000770f8c95ed97/staging/src/k8s.io/api/batch/v1/types.go#L421-L425) where SSA uses atomic.

Comment (liggitt) [reply contributors/devel/sig-architecture/api-conventions.md]: yeah, a sweep is probably good

File: contributors/devel/sig-architecture/api-conventions.md [hunk 2]
```diff
@@ -425,7 +425,10 @@ Conditions are most useful when they follow some consistent conventions:
 Conditions should follow the standard schema included in [k8s.io/apimachinery/pkg/apis/meta/v1/types.go](https://github.com/kubernetes/apimachinery/blob/release-1.23/pkg/apis/meta/v1/types.go#L1432-L1492).
 It should be included as a top level element in status, similar to
 ```go
-Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+// +listType=map
+// +listMapKey=type
```
Comment (liggitt) [inline contributors/devel/sig-architecture/api-conventions.md:429]: I would include these as well to keep client-side apply working as we want
```go
	// +patchStrategy=merge
	// +patchMergeKey=type
```

Comment (jpbetz) [reply contributors/devel/sig-architecture/api-conventions.md:429]: Done. I'll do a sweep separately.

**Comment by jpbetz** [comment]
/assign @johnbelamaric 
For approval.

**Review by johnbelamaric — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**